File: app/src/main/java/com/wang/example/msg/adapter/ImgAdapter.java
package com.wang.example.msg.adapter;

import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

import com.wang.container.adapter.OneContainerItemAdapter;
import com.wang.container.holder.BaseViewHolder;
import com.wang.container.interfaces.OnItemClickListener;
import com.wang.example.msg.bean.ImgBean;
import com.wang.example.utils.ToastUtils;

public class ImgAdapter extends OneContainerItemAdapter<ViewDataBinding, ImgBean> {

    public ImgAdapter() {
        setOnItemClickListener(new OnItemClickListener<ImgBean>() {
            @Override
            public void onItemClick(@NonNull View view, int position) {
                ToastUtils.toast("您点击了图片，绝对位置：" + getViewHolder(view).getCommonPosition());
            }
        });
    }

    @Override
    protected BaseViewHolder<ViewDataBinding> onCreateChildViewHolder(ViewGroup parent) {
        AppCompatImageView iv = new AppCompatImageView(parent.getContext());
        iv.setLayoutParams(new RecyclerView.LayoutParams(300, 300));
        iv.setPadding(0, 80, 0, 80);
        return new BaseViewHolder<>(iv);
    }

    @Override
    protected void onBindChildViewHolder(@NonNull BaseViewHolder<ViewDataBinding> holder, ImgBean bean) {
        ImageView iv = (ImageView) holder.itemView;
        iv.setImageResource(getCurrentBean().imgInfo.imgRes);
    }
}

File: app/src/main/java/com/wang/example/msg/adapter/ImgAdapter.kt
package com.wang.example.msg.adapter

import android.view.ViewGroup
import android.widget.ImageView
import androidx.appcompat.widget.AppCompatImageView
import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding
import com.wang.container.adapter.OneContainerItemAdapter
import com.wang.container.holder.BaseViewHolder
import com.wang.example.msg.bean.ImgBean
import com.wang.example.utils.toast

class ImgAdapter : OneContainerItemAdapter<ViewBinding, ImgBean>() {
    override fun onCreateChildViewHolder(parent: ViewGroup): BaseViewHolder<ViewBinding> {
        val iv = AppCompatImageView(parent.context)
        iv.layoutParams = RecyclerView.LayoutParams(300, 300)
        iv.setPadding(0, 80, 0, 80)
        return BaseViewHolder(iv)
    }

    override fun onBindChildViewHolder(
        holder: BaseViewHolder<ViewBinding>,
        currentBean: ImgBean
    ) {
        val iv = holder.itemView as ImageView
        iv.setImageResource(currentBean.imgInfo.imgRes)
    }

    init {
        setOnItemClickListener { _, _, _, vh, _, _ ->
            "您点击了图片，绝对位置：${vh.commonPosition}".toast()
        }

        setOnItemLongClickListener { _, _, _, vh, _, _ ->
            "您长按了图片，绝对位置：${vh.commonPosition}".toast()
            true
        }
    }
}